### PR TITLE
Add more examples

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

Adds a few more examples:
- Imitation of a camera viewfinder with filters, pinch-to-zoom, and option to long-press the shutter button to record video
- Imitation of a calculator app with expandable side panels and scrollable history
- Simple transformations demo
-  Test for overlapping elements

## Test plan

Tested new examples on the Example app.

### CODE REVIEW

File: examples/Example/src/new_api/calculator/index.tsx [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+  scrollView.current?.scrollToEnd({ animated: true });
+
+  return (
+    <GestureMonitor gesture={dragGesture}>
+      <Animated.View
+        style={[styles.output, translationStyle]}
+        onLayout={measure}>
+        <ScrollView
+          ref={(ref: ScrollView) => {
+            if (!opened && ref) ref.scrollToEnd({ animated: false });
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:130]:
```suggestion
            if (!opened && ref) { ref.scrollToEnd({ animated: false }); }
```

File: examples/Example/src/new_api/calculator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+        </ScrollView>
+        <View style={styles.handleView}>
+          <View style={styles.handle} />
+        </View>
+      </Animated.View>
+    </GestureMonitor>
+  );
+}
+
+function Expression(props: { expression: string }) {
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:151]:
```suggestion
function Expression({ expression }: { expression: string }) {
```

File: examples/Example/src/new_api/calculator/index.tsx [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+  return (
+    <View style={styles.expression}>
+      <View style={{ flexDirection: 'column' }}>
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:154]: Can we merge them together?

File: examples/Example/src/new_api/calculator/index.tsx [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+  );
+}
+
+function Input(props: {
+  setHistory: Dispatch<SetStateAction<string[]>>;
+  setExpression: Dispatch<SetStateAction<string>>;
+  measure: (e: LayoutChangeEvent) => void;
+  offset: Animated.SharedValue<number>;
+  expression: string;
+}) {
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:168]:
```suggestion
interface InputProps {
  setHistory: Dispatch<SetStateAction<string[]>>;
  setExpression: Dispatch<SetStateAction<string>>;
  measure: (e: LayoutChangeEvent) => void;
  offset: Animated.SharedValue<number>;
  expression: string;
}

function Input({ setHistory, setExpression, measure, offset, expression }: InputProps) {
```

File: examples/Example/src/new_api/calculator/index.tsx [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+  const translationStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: props.offset.value }],
+    };
+  });
+
+  function append(symbol: string) {
+    if (symbol === '<') {
+      props.setHistory((h) => [...h, props.expression]);
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:177]: `h.concat(expression)`

File: examples/Example/src/new_api/calculator/index.tsx [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+        const value = dragOffset.value + e.translationX;
+
+        if (opened) {
+          if (value > -layout.value.width + margin + 75) {
+            runOnJS(close)();
+          } else {
+            runOnJS(open)();
+          }
+        } else {
+          if (value < -75) {
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:261]: Extract it to constant.

File: examples/Example/src/new_api/camera/index.tsx [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+  const [selectedFilter, setSelectedFilter] = useState(0);
+  const [isRecording, setIsRecording] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(MAX_VIDEO_DURATION);
+  const [recordingIntervalHandle, setRecordingIntervalHandle] = useState(-1);
+
+  const filtersPanGesture = Gesture.pan()
+    .setOnUpdate((e) => {
+      'worklet';
+      filter.value =
+        filter.value + (filterOffset.value - e.translationX) * 0.01;
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/camera/index.tsx:34]:
```suggestion
        filter.value + (filterOffset.value - e.translationX) / 100;
```

File: examples/Example/src/new_api/camera/index.tsx [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+    );
+  }
+
+  function finishRecording() {
+    setIsRecording(false);
+    clearInterval(recordingIntervalHandle);
+    setRemainingTime(MAX_VIDEO_DURATION);
+
+    Alert.alert(
+      `You took a video (${(MAX_VIDEO_DURATION - remainingTime) * 0.001} s)`
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/camera/index.tsx:143]:
```suggestion
      `You took a video (${(MAX_VIDEO_DURATION - remainingTime) / 1000} s)`
```

File: examples/Example/src/new_api/camera/index.tsx [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+  filter: string;
+  selected: Animated.SharedValue<number>;
+}) {
+  const window = useWindowDimensions();
+  const style = useAnimatedStyle(() => {
+    return {
+      transform: [
+        {
+          translateX:
+            window.width * 0.5 -
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/camera/index.tsx:229]:
```suggestion
            window.width / 2 -
```

File: examples/Example/src/new_api/overlap_parent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+} from 'react-native-gesture-handler';
+
+function Box(props: {
+  color: string;
+  overlap?: boolean;
+  children?: React.ReactNode;
+}) {
+  const gesture = useGesture(
+    Gesture.tap().setOnEnd((_e, success) => {
+      if (success) console.log(props.color);
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/overlap_parent/index.tsx:16]:
```suggestion
      if (success) { console.log(props.color) };
```

File: examples/Example/src/new_api/camera/index.tsx [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+} from 'react-native-gesture-handler';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  runOnJS,
+} from 'react-native-reanimated';
+
+const filters = ['red', 'green', 'blue', 'yellow', 'orange', 'cyan'];
+const MAX_VIDEO_DURATION = 60_000;
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/camera/index.tsx:16]:
```suggestion
const MAX_VIDEO_DURATION_MS = 60_000;
```

File: examples/Example/src/new_api/camera/index.tsx [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+const FILTER_BUTTON_RADIUS = 35;
+
+export default function Home() {
+  const filter = useSharedValue(0);
+  const filterOffset = useSharedValue(0);
+  const zoom = useSharedValue(1);
+  const [scale, setScale] = useState(1);
+  const [selectedFilter, setSelectedFilter] = useState(0);
+  const [isRecording, setIsRecording] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(MAX_VIDEO_DURATION);
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/camera/index.tsx:27]:
```suggestion
  const [remainingTimeMs, setRemainingTimeMs] = useState(MAX_VIDEO_DURATION_MS);
```

**Review by jakub-gonet — CHANGES_REQUESTED** [review]
Only mentioned a few first occurrences, please everything that applies.

File: examples/Example/src/App.tsx [hunk 13]
```diff
@@ -117,6 +141,14 @@ export default function App() {
             options={{ title: SCREENS[name].title || name }}
           />
         ))}
+        {Object.keys(NEW_API_SCREENS).map((name) => (
+          <Stack.Screen
+            key={name}
+            name={name}
+            getComponent={() => NEW_API_SCREENS[name].component}
+            options={{ title: NEW_API_SCREENS[name].title || name }}
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/App.tsx]: Fixed in bb90e7cca7724a2a8b609faa20340c0d7b2b0369

File: examples/Example/src/App.tsx [hunk 14]
```diff
@@ -145,6 +177,28 @@ function MainScreen({ navigation }: StackScreenProps<ParamListBase>) {
   );
 }
 
+function NewApiScreen({ navigation }: StackScreenProps<ParamListBase>) {
+  const data = Object.keys(NEW_API_SCREENS).map((key) => {
+    const item = NEW_API_SCREENS[key];
+    return { key, title: item.title || key };
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/App.tsx]: Fixed in bb90e7cca7724a2a8b609faa20340c0d7b2b0369

File: examples/Example/src/new_api/calculator/index.tsx [hunk 15]
```diff
@@ -0,0 +1,399 @@
...
+const window = Dimensions.get('window');
+
+export default function CalculatorUI() {
+  const outputOffset = useSharedValue(0);
+  const [history, setHistory] = useState(Array<string>());
+  const [expression, setExpression] = useState('');
+
+  function measure(e: LayoutChangeEvent) {
+    outputOffset.value = -e.nativeEvent.layout.height;
+  }
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 6d7883bbc600effc3ea23d308d3e6775a478cb6b

File: examples/Example/src/new_api/calculator/index.tsx [hunk 16]
```diff
@@ -0,0 +1,399 @@
...
+}) {
+  const layout = useRef({});
+  const scrollView = useRef<ScrollView>();
+  const drag = useSharedValue(0);
+  const dragOffset = useSharedValue(0);
+  const [opened, setOpened] = useState(false);
+
+  function measure(e: LayoutChangeEvent) {
+    layout.current = e.nativeEvent.layout;
+  }
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 6d7883bbc600effc3ea23d308d3e6775a478cb6b

File: examples/Example/src/new_api/calculator/index.tsx [hunk 17]
```diff
@@ -0,0 +1,399 @@
...
+      />
+    </View>
+  );
+}
+
+function Output(props: {
+  offset: Animated.SharedValue<number>;
+  expression: string;
+  history: string[];
+}) {
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 4cb994424fd9597c29bc5210309f1e75b80b9a55

File: examples/Example/src/new_api/calculator/index.tsx [hunk 18]
```diff
@@ -0,0 +1,399 @@
...
+    return {
+      transform: [{ translateY: props.offset.value + drag.value }],
+    };
+  });
+
+  const dragGesture = useAnimatedGesture(
+    Gesture.pan()
+      .setOnUpdate((e) => {
+        'worklet';
+        const value = dragOffset.value + e.translationY;
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Renamed in aa89ad2924cca02b124529030acb1b96dbe47b32

File: examples/Example/src/new_api/calculator/index.tsx (hunk 1 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Fixed in 3a5fca0b55a2d504c90c4cb9449f48ce4ebefced

File: examples/Example/src/new_api/calculator/index.tsx (hunk 2 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 3a5fca0b55a2d504c90c4cb9449f48ce4ebefced

File: examples/Example/src/new_api/calculator/index.tsx (hunk 3 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in f31ff9115a22a1858f30f24b2707fb3e8a2b9c65

File: examples/Example/src/new_api/calculator/index.tsx (hunk 4 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in ad1e9b516cb93f68ab2b54e73900dafcc220646c

File: examples/Example/src/new_api/calculator/index.tsx (hunk 5 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 26c0974724e12b9c9e06ce531674471ac46d05c3

File: examples/Example/src/new_api/calculator/index.tsx (hunk 6 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/calculator/index.tsx]: Done in 26c0974724e12b9c9e06ce531674471ac46d05c3

File: examples/Example/src/new_api/camera/index.tsx (hunk 7 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/camera/index.tsx]: Done in c9c43d8757b585808251376d16f6a0182c2842f3

File: examples/Example/src/new_api/camera/index.tsx (hunk 8 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/camera/index.tsx]: Done in c9c43d8757b585808251376d16f6a0182c2842f3

File: examples/Example/src/new_api/camera/index.tsx (hunk 9 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/camera/index.tsx]: Done in c9c43d8757b585808251376d16f6a0182c2842f3

File: examples/Example/src/new_api/overlap_parent/index.tsx (hunk 10 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/overlap_parent/index.tsx]: Done in 1e19778e795f293d5f933ab6dfa528b78373a64f

File: examples/Example/src/new_api/camera/index.tsx (hunk 11 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/camera/index.tsx]: Done in ae78c347855f8ebb233e8fa59d2e0794bc4691a3

File: examples/Example/src/new_api/camera/index.tsx (hunk 12 above)
Comment (j-piasecki) [reply examples/Example/src/new_api/camera/index.tsx]: Done in ae78c347855f8ebb233e8fa59d2e0794bc4691a3

File: examples/Example/src/new_api/calculator/index.tsx [hunk 19]
```diff
@@ -0,0 +1,429 @@
...
+  return (
+    <GestureMonitor gesture={dragGesture}>
+      <Animated.View
+        style={[styles.output, translationStyle]}
+        onLayout={measure}>
+        <ScrollView
+          ref={(ref: ScrollView) => {
+            if (!opened && ref) {
+              ref.scrollToEnd({ animated: false });
+            }
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/calculator/index.tsx:140]:
```suggestion
            if (!opened) {
              ref?.scrollToEnd({ animated: false });
            }
```

**Review by jakub-gonet — APPROVED** [review]
Fix conflicts and we're good to go.

**Pull request merged**